### PR TITLE
Add `RequestResult` type

### LINKED ISSUE #477: Change `Result` type in batching requests methods
### Feature Request

Currently `JsonRpcProvider` methods for batching requests (`batchRequests` and `batchRequestsAny`) return values wrapped in kotlin's `Result`. It works in Kotlin but there is no support for it in `Java`, therefore we cannot access result values. `Result` needs to be changed to a custom type.

### PR DESCRIPTION
**Pull request by franciszekjob**
## Describe your changes

- Add custom type `RequestResult` which is a wrapper for kotlin's `Result`

## Linked issues

Closes #477 

## Breaking changes

- [ ] This issue contains breaking changes

### CODE REVIEW

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetResult.kt
Comment (DelevoXDG) [inline lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetResult.kt:9]: Let's keep method naming the same as in [Kotlin Result](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/), otherwise it seems a bit confusing to me.

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetResult.kt
Comment (DelevoXDG) [inline lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetResult.kt:7]: Why not use Kotlin `Result` under the hood, and avoid implementing any logic ourselves?

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetResult.kt
Comment (DelevoXDG) [inline lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetResult.kt:3]: Nit:|

Not sure about this, but maybe `RequestResult`?

**Review by ddoktorski — APPROVED** [review]
LGTM, but let's wait for final approval from @DelevoXDG

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/RequestResult.kt
Comment (franciszekjob) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/RequestResult.kt]: Actually in `failure` we have to keep `RequestResult<T>`, as there is not enough information to infer the type.

**Pull request merged**
